taipy/core/data/data_node.py

```diff
@@ -346,6 +348,21 @@ def __getstate__(self):
     def __setstate__(self, state):
         vars(self).update(state)
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        if self.__CHECK_INIT_DONE_ATTR_NAME not in dir(self) or name in dir(self):
+            return super().__setattr__(name, value)
+        else:
+            protected_attribute_name = _validate_id(name)
+            try:
+                if protected_attribute_name not in self._properties:
+                    raise AttributeError
+                raise AttributeKeyAlreadyExisted(name)
+            except AttributeError:
+                return super().__setattr__(name, value)
```

Comment:
Does this work?
Raise and catch right away seems weird to me.
```suggestion
            if protected_attribute_name in self._properties:
                raise AttributeKeyAlreadyExisted(name)
            return super().__setattr__(name, value)
```